File: backend/app/schemas/trainer.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import AlertType, DietPreference, GoalType
from app.schemas.common import ORMModel
from app.schemas.member import MemberProfileRead, WorkoutLogRead


class TrainerAssignedMemberRead(BaseModel):
    user_id: UUID
    full_name: str
    email: str
    goal: GoalType | None
    diet_preference: DietPreference | None
    last_workout_at: datetime | None
    workouts_completed: int
    is_at_risk: bool


class TrainerMemberDetailRead(BaseModel):
    member: TrainerAssignedMemberRead
    profile: MemberProfileRead | None
    recent_workouts: list[WorkoutLogRead]


class TrainerNoteCreate(BaseModel):
    note: str


class TrainerNoteRead(ORMModel):
    id: UUID
    trainer_id: UUID
    member_id: UUID
    note: str
    created_at: datetime
    updated_at: datetime


class ActivityAlertRead(ORMModel):
    id: UUID
    user_id: UUID
    gym_id: UUID
    alert_type: AlertType
    message: str
    is_read: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/trainer.ts
import type { MemberProfile, WorkoutLog } from "./member";

export interface TrainerAssignedMember {
  user_id: string;
  full_name: string;
  email: string;
  goal: "lose" | "gain" | "maintain" | null;
  diet_preference: "vegetarian" | "non_vegetarian" | null;
  last_workout_at: string | null;
  workouts_completed: number;
  is_at_risk: boolean;
}

export interface TrainerMemberDetail {
  member: TrainerAssignedMember;
  profile: MemberProfile | null;
  recent_workouts: WorkoutLog[];
}

export interface TrainerNote {
  id: string;
  trainer_id: string;
  member_id: string;
  note: string;
  created_at: string;
  updated_at: string;
}

export interface ActivityAlert {
  id: string;
  user_id: string;
  gym_id: string;
  alert_type: "inactive" | "goal_stalled" | "new_signup" | "streak_broken";
  message: string;
  is_read: boolean;
  created_at: string;
  updated_at: string;
}
